nessie/src/main/java/org/apache/iceberg/nessie/UpdateableReference.java

```diff
@@ -44,6 +45,15 @@ public boolean refresh() throws NessieNotFoundException {
     return !oldReference.equals(reference);
   }
 
+  public void updateReference(Reference ref) {
+    Objects.requireNonNull(ref);
+    if (this.reference != null && this.reference.getClass() != ref.getClass()) {
```

Comment:
can `this.reference` ever be `null`?

Reply:
Probably not.
Added some `@Nonnull` and removed that check.